Chad's branch

cpeters008 opened the pull request:
Merge my branch with master

cpeters008 committed:
Update common.h

1. #include <limits> - included for error checking user input ranges
2. Added constants for menu option handling and user char input error checking
3. struct chnArray - added for chained hash table. This will allow the array to have a pointer in each cell to be used to point to hash data via separate chaining.

cpeters008 committed:
Update HashADT.cpp

1. InitializeOpenTbl - initializes array and index array cells all to 0. Ref. pg. 519-520 in our book for why the index array is nice. I added it so that the searching / finding data in hash table will be better managed / easier.
2. InitializeChnTbl - initializes array of pointers to NULL

cpeters008 committed:
Update HashADT.h

1. InitializeOpenTbl - prototype for function
2. InitializeChnTbl - prototype for function

cpeters008 committed:
Update OtherFunctions.cpp

1. int GetMenuChoice(); - Displays menu, error checks, returns menu choice 
2. int GetTableSize(); - Gets table size from user, error checks, returns table size
3. char KeepTesting(); - Gets char input from user, error  checks, returns Y or N (Y for continue and N for quit program)

cpeters008 committed:
Update OtherFunctions.h

1. int GetMenuChoice(); - prototype
2. int GetTableSize(); - prototype
3. char KeepTesting(); - prototype

cpeters008 committed:
derp add

Re-added your KnuthAvh ... I think I overwrote it by accident .. BUT ITS BACK NOW -- DON'T WORRY!

cpeters008 committed:
Update main.cpp

1. Created initial data structure for handling user input and processing of information for project
2. Added the following 7 variables, error checking, and debug statements. Reference changes to see them all in more detail.

cpeters008 committed:
Update Common.h

1. I found a duplicate constant and deleted it.
2. Added libraries <limits>, <cstdlib>, and <ctime> - limits is needed for my error checking of user input, etc.

cpeters008 committed:
Update HashADT.cpp

1. Added your changes

cpeters008 committed:
Update HashADT.h

Added your changes

cpeters008 committed:
Update OtherFunctions.cpp

1. Added functions RandomNum(), DuplicateValue(), and InitializeRandomArray().
--RandomNum utilizes a slightly more random formula
--DuplicateValue utilizes less time to search each time by utilizing the counter from within the for loop as a parameter
--InitializeRandomArray accounts  for when the array is initially empty and ensuring no cells are skipped

cpeters008 committed:
Update OtherFunctions.h

Added new functions as headers

cpeters008 committed:
Update main.cpp

Updated to mirror how yours looked.

cpeters008 committed:
Included your functions and QuadraticProbe / DoubleHashValue

--I have not tested either function yet to see if they work.. I have to go to work now, but I'll test them there

cpeters008 committed:
Added your  functions abd QuadraticProbe / DoubleHashValue

--Both of my functions still require testing

cpeters008 committed:
3 new consts added

--consts for QuadraticProbe and DoubleHashValue added

faust06 commented:
Alrighty, I’ll get on it. It isn’t able to be merged directly online. It’ll have to be done via the command line. I’ll merge and let you know when it’s done.

-Neil

> On Mar 27, 2015, at 11:11 AM, Chad notifications@github.com wrote:
> 
> Merge my branch with master
> 
> You can view, comment on, or merge this pull request online at:
> 
>   https://github.com/faust06/Week4-Assn4/pull/6 https://github.com/faust06/Week4-Assn4/pull/6
> Commit Summary
> 
> Update common.h
> Update HashADT.cpp
> Update HashADT.h
> Update OtherFunctions.cpp
> Update OtherFunctions.h
> derp add
> Update main.cpp
> Update Common.h
> Update HashADT.cpp
> Update HashADT.h
> Update OtherFunctions.cpp
> Update OtherFunctions.h
> Update main.cpp
> Included your functions and QuadraticProbe / DoubleHashValue
> Added your functions abd QuadraticProbe / DoubleHashValue
> 3 new consts added
> File Changes
> 
> M townsendpeters-prog/Common.h https://github.com/faust06/Week4-Assn4/pull/6/files#diff-0 (24)
> M townsendpeters-prog/HashADT.cpp https://github.com/faust06/Week4-Assn4/pull/6/files#diff-1 (278)
> M townsendpeters-prog/HashADT.h https://github.com/faust06/Week4-Assn4/pull/6/files#diff-2 (8)
> M townsendpeters-prog/OtherFunctions.cpp https://github.com/faust06/Week4-Assn4/pull/6/files#diff-3 (286)
> M townsendpeters-prog/OtherFunctions.h https://github.com/faust06/Week4-Assn4/pull/6/files#diff-4 (7)
> M townsendpeters-prog/main.cpp https://github.com/faust06/Week4-Assn4/pull/6/files#diff-5 (79)
> Patch Links:
> 
> https://github.com/faust06/Week4-Assn4/pull/6.patch https://github.com/faust06/Week4-Assn4/pull/6.patch
> https://github.com/faust06/Week4-Assn4/pull/6.diff https://github.com/faust06/Week4-Assn4/pull/6.diff
> —
> Reply to this email directly or view it on GitHub https://github.com/faust06/Week4-Assn4/pull/6.

cpeters008 committed:
Line 30 changed ; to , syntax error

cpeters008 commented:
I'm going to start testing and troubleshooting the doublehash and quadratic
probes.

On Fri, Mar 27, 2015 at 7:12 PM, faust06 notifications@github.com wrote:

> Alrighty, I'll get on it. It isn't able to be merged directly online.
> It'll have to be done via the command line. I'll merge and let you know
> when it's done.
> 
> -Neil
> 
> > On Mar 27, 2015, at 11:11 AM, Chad notifications@github.com wrote:
> > 
> > Merge my branch with master
> > 
> > You can view, comment on, or merge this pull request online at:
> > 
> > https://github.com/faust06/Week4-Assn4/pull/6 <
> > https://github.com/faust06/Week4-Assn4/pull/6>
> > Commit Summary
> > 
> > Update common.h
> > Update HashADT.cpp
> > Update HashADT.h
> > Update OtherFunctions.cpp
> > Update OtherFunctions.h
> > derp add
> > Update main.cpp
> > Update Common.h
> > Update HashADT.cpp
> > Update HashADT.h
> > Update OtherFunctions.cpp
> > Update OtherFunctions.h
> > Update main.cpp
> > Included your functions and QuadraticProbe / DoubleHashValue
> > Added your functions abd QuadraticProbe / DoubleHashValue
> > 3 new consts added
> > File Changes
> > 
> > M townsendpeters-prog/Common.h <
> > https://github.com/faust06/Week4-Assn4/pull/6/files#diff-0> (24)
> > M townsendpeters-prog/HashADT.cpp <
> > https://github.com/faust06/Week4-Assn4/pull/6/files#diff-1> (278)
> > M townsendpeters-prog/HashADT.h <
> > https://github.com/faust06/Week4-Assn4/pull/6/files#diff-2> (8)
> > M townsendpeters-prog/OtherFunctions.cpp <
> > https://github.com/faust06/Week4-Assn4/pull/6/files#diff-3> (286)
> > M townsendpeters-prog/OtherFunctions.h <
> > https://github.com/faust06/Week4-Assn4/pull/6/files#diff-4> (7)
> > M townsendpeters-prog/main.cpp <
> > https://github.com/faust06/Week4-Assn4/pull/6/files#diff-5> (79)
> > Patch Links:
> > 
> > https://github.com/faust06/Week4-Assn4/pull/6.patch <
> > https://github.com/faust06/Week4-Assn4/pull/6.patch>
> > https://github.com/faust06/Week4-Assn4/pull/6.diff <
> > 
> > ## https://github.com/faust06/Week4-Assn4/pull/6.diff>
> > 
> > Reply to this email directly or view it on GitHub <
> > https://github.com/faust06/Week4-Assn4/pull/6>.
> 
> ## 
> 
> Reply to this email directly or view it on GitHub
> https://github.com/faust06/Week4-Assn4/pull/6#issuecomment-87037585.

cpeters008 committed:
Added HashValue function

cpeters008 commented:
Hey,

I want to go ahead and create an insertHashVal() function. Have you started
or created one yet?

Chad

On Fri, Mar 27, 2015 at 7:12 PM, faust06 notifications@github.com wrote:

> Alrighty, I'll get on it. It isn't able to be merged directly online.
> It'll have to be done via the command line. I'll merge and let you know
> when it's done.
> 
> -Neil
> 
> > On Mar 27, 2015, at 11:11 AM, Chad notifications@github.com wrote:
> > 
> > Merge my branch with master
> > 
> > You can view, comment on, or merge this pull request online at:
> > 
> > https://github.com/faust06/Week4-Assn4/pull/6 <
> > https://github.com/faust06/Week4-Assn4/pull/6>
> > Commit Summary
> > 
> > Update common.h
> > Update HashADT.cpp
> > Update HashADT.h
> > Update OtherFunctions.cpp
> > Update OtherFunctions.h
> > derp add
> > Update main.cpp
> > Update Common.h
> > Update HashADT.cpp
> > Update HashADT.h
> > Update OtherFunctions.cpp
> > Update OtherFunctions.h
> > Update main.cpp
> > Included your functions and QuadraticProbe / DoubleHashValue
> > Added your functions abd QuadraticProbe / DoubleHashValue
> > 3 new consts added
> > File Changes
> > 
> > M townsendpeters-prog/Common.h <
> > https://github.com/faust06/Week4-Assn4/pull/6/files#diff-0> (24)
> > M townsendpeters-prog/HashADT.cpp <
> > https://github.com/faust06/Week4-Assn4/pull/6/files#diff-1> (278)
> > M townsendpeters-prog/HashADT.h <
> > https://github.com/faust06/Week4-Assn4/pull/6/files#diff-2> (8)
> > M townsendpeters-prog/OtherFunctions.cpp <
> > https://github.com/faust06/Week4-Assn4/pull/6/files#diff-3> (286)
> > M townsendpeters-prog/OtherFunctions.h <
> > https://github.com/faust06/Week4-Assn4/pull/6/files#diff-4> (7)
> > M townsendpeters-prog/main.cpp <
> > https://github.com/faust06/Week4-Assn4/pull/6/files#diff-5> (79)
> > Patch Links:
> > 
> > https://github.com/faust06/Week4-Assn4/pull/6.patch <
> > https://github.com/faust06/Week4-Assn4/pull/6.patch>
> > https://github.com/faust06/Week4-Assn4/pull/6.diff <
> > 
> > ## https://github.com/faust06/Week4-Assn4/pull/6.diff>
> > 
> > Reply to this email directly or view it on GitHub <
> > https://github.com/faust06/Week4-Assn4/pull/6>.
> 
> ## 
> 
> Reply to this email directly or view it on GitHub
> https://github.com/faust06/Week4-Assn4/pull/6#issuecomment-87037585.

faust06 commented:
I haven't started one yet. I won't be able to work until I get off work this afternoon

> On Mar 27, 2015, at 11:46 AM, Chad notifications@github.com wrote:
> 
> Hey,
> 
> I want to go ahead and create an insertHashVal() function. Have you started
> or created one yet?
> 
> Chad
> 
> On Fri, Mar 27, 2015 at 7:12 PM, faust06 notifications@github.com wrote:
> 
> > Alrighty, I'll get on it. It isn't able to be merged directly online.
> > It'll have to be done via the command line. I'll merge and let you know
> > when it's done.
> > 
> > -Neil
> > 
> > > On Mar 27, 2015, at 11:11 AM, Chad notifications@github.com wrote:
> > > 
> > > Merge my branch with master
> > > 
> > > You can view, comment on, or merge this pull request online at:
> > > 
> > > https://github.com/faust06/Week4-Assn4/pull/6 <
> > > https://github.com/faust06/Week4-Assn4/pull/6>
> > > Commit Summary
> > > 
> > > Update common.h
> > > Update HashADT.cpp
> > > Update HashADT.h
> > > Update OtherFunctions.cpp
> > > Update OtherFunctions.h
> > > derp add
> > > Update main.cpp
> > > Update Common.h
> > > Update HashADT.cpp
> > > Update HashADT.h
> > > Update OtherFunctions.cpp
> > > Update OtherFunctions.h
> > > Update main.cpp
> > > Included your functions and QuadraticProbe / DoubleHashValue
> > > Added your functions abd QuadraticProbe / DoubleHashValue
> > > 3 new consts added
> > > File Changes
> > > 
> > > M townsendpeters-prog/Common.h <
> > > https://github.com/faust06/Week4-Assn4/pull/6/files#diff-0> (24)
> > > M townsendpeters-prog/HashADT.cpp <
> > > https://github.com/faust06/Week4-Assn4/pull/6/files#diff-1> (278)
> > > M townsendpeters-prog/HashADT.h <
> > > https://github.com/faust06/Week4-Assn4/pull/6/files#diff-2> (8)
> > > M townsendpeters-prog/OtherFunctions.cpp <
> > > https://github.com/faust06/Week4-Assn4/pull/6/files#diff-3> (286)
> > > M townsendpeters-prog/OtherFunctions.h <
> > > https://github.com/faust06/Week4-Assn4/pull/6/files#diff-4> (7)
> > > M townsendpeters-prog/main.cpp <
> > > https://github.com/faust06/Week4-Assn4/pull/6/files#diff-5> (79)
> > > Patch Links:
> > > 
> > > https://github.com/faust06/Week4-Assn4/pull/6.patch <
> > > https://github.com/faust06/Week4-Assn4/pull/6.patch>
> > > https://github.com/faust06/Week4-Assn4/pull/6.diff <
> > > 
> > > ## https://github.com/faust06/Week4-Assn4/pull/6.diff>
> > > 
> > > Reply to this email directly or view it on GitHub <
> > > https://github.com/faust06/Week4-Assn4/pull/6>.
> > 
> > ## 
> > 
> > Reply to this email directly or view it on GitHub
> > https://github.com/faust06/Week4-Assn4/pull/6#issuecomment-87037585.
> > 
> > —
> > Reply to this email directly or view it on GitHub.

cpeters008 commented:
Ok,

I'm going to be working on the program for awhile (almost all night if I
can). right now I tested the Quadraticprobing and it works well. Let me
know when you're home. I'll update everything on github in my branch for
you to review what I've been up to. Then we can talk about it and hopefully
merge into the main and press on. My goal is to ensure that all the
collision resolution, hash table filling from the random array with unique
numbers, duplicate search, all work perfect.

I'm not going to worry about the destroy() function because you said you
were going to make that one.

enjoy your work :D.

On Fri, Mar 27, 2015 at 8:06 PM, faust06 notifications@github.com wrote:

> I haven't started one yet. I won't be able to work until I get off work
> this afternoon
> 
> > On Mar 27, 2015, at 11:46 AM, Chad notifications@github.com wrote:
> > 
> > Hey,
> > 
> > I want to go ahead and create an insertHashVal() function. Have you
> > started
> > or created one yet?
> > 
> > Chad
> > 
> > On Fri, Mar 27, 2015 at 7:12 PM, faust06 notifications@github.com
> > wrote:
> > 
> > > Alrighty, I'll get on it. It isn't able to be merged directly online.
> > > It'll have to be done via the command line. I'll merge and let you know
> > > when it's done.
> > > 
> > > -Neil
> > > 
> > > > On Mar 27, 2015, at 11:11 AM, Chad notifications@github.com wrote:
> > > > 
> > > > Merge my branch with master
> > > > 
> > > > You can view, comment on, or merge this pull request online at:
> > > > 
> > > > https://github.com/faust06/Week4-Assn4/pull/6 <
> > > > https://github.com/faust06/Week4-Assn4/pull/6>
> > > > Commit Summary
> > > > 
> > > > Update common.h
> > > > Update HashADT.cpp
> > > > Update HashADT.h
> > > > Update OtherFunctions.cpp
> > > > Update OtherFunctions.h
> > > > derp add
> > > > Update main.cpp
> > > > Update Common.h
> > > > Update HashADT.cpp
> > > > Update HashADT.h
> > > > Update OtherFunctions.cpp
> > > > Update OtherFunctions.h
> > > > Update main.cpp
> > > > Included your functions and QuadraticProbe / DoubleHashValue
> > > > Added your functions abd QuadraticProbe / DoubleHashValue
> > > > 3 new consts added
> > > > File Changes
> > > > 
> > > > M townsendpeters-prog/Common.h <
> > > > https://github.com/faust06/Week4-Assn4/pull/6/files#diff-0> (24)
> > > > M townsendpeters-prog/HashADT.cpp <
> > > > https://github.com/faust06/Week4-Assn4/pull/6/files#diff-1> (278)
> > > > M townsendpeters-prog/HashADT.h <
> > > > https://github.com/faust06/Week4-Assn4/pull/6/files#diff-2> (8)
> > > > M townsendpeters-prog/OtherFunctions.cpp <
> > > > https://github.com/faust06/Week4-Assn4/pull/6/files#diff-3> (286)
> > > > M townsendpeters-prog/OtherFunctions.h <
> > > > https://github.com/faust06/Week4-Assn4/pull/6/files#diff-4> (7)
> > > > M townsendpeters-prog/main.cpp <
> > > > https://github.com/faust06/Week4-Assn4/pull/6/files#diff-5> (79)
> > > > Patch Links:
> > > > 
> > > > https://github.com/faust06/Week4-Assn4/pull/6.patch <
> > > > https://github.com/faust06/Week4-Assn4/pull/6.patch>
> > > > https://github.com/faust06/Week4-Assn4/pull/6.diff <
> > > > 
> > > > ## https://github.com/faust06/Week4-Assn4/pull/6.diff>
> > > > 
> > > > Reply to this email directly or view it on GitHub <
> > > > https://github.com/faust06/Week4-Assn4/pull/6>.
> > > 
> > > ## 
> > > 
> > > Reply to this email directly or view it on GitHub
> > > https://github.com/faust06/Week4-Assn4/pull/6#issuecomment-87037585.
> > > 
> > > ## 
> > > 
> > > Reply to this email directly or view it on GitHub.
> 
> ## 
> 
> Reply to this email directly or view it on GitHub
> https://github.com/faust06/Week4-Assn4/pull/6#issuecomment-87059686.

cpeters008 commented:
Hey, so during testing I noticed that the random numbers assigned from the
randomArray[] to openHashTbl[] sometimes are the same as the array position
in the openHashTbl[].

for example, number 11 from randomArray[] will be stored in openHashTbl[11].

If you stay in the program and continue to re-run the same test (I'm
testing the quadratic probing at the moment) the number of identical
numbers to their respective cell numbers increases by 1 every time.
Initially, the identical numbers start out at about 926 out of 6700 and
after each test climb by 1 and theoretically would eventually reach 6700 if
ran enough times.

My question here that I need help with is... whats wrong? Is this supposed
to happen? I'll post my debug data to github

On Fri, Mar 27, 2015 at 8:06 PM, faust06 notifications@github.com wrote:

> I haven't started one yet. I won't be able to work until I get off work
> this afternoon
> 
> > On Mar 27, 2015, at 11:46 AM, Chad notifications@github.com wrote:
> > 
> > Hey,
> > 
> > I want to go ahead and create an insertHashVal() function. Have you
> > started
> > or created one yet?
> > 
> > Chad
> > 
> > On Fri, Mar 27, 2015 at 7:12 PM, faust06 notifications@github.com
> > wrote:
> > 
> > > Alrighty, I'll get on it. It isn't able to be merged directly online.
> > > It'll have to be done via the command line. I'll merge and let you know
> > > when it's done.
> > > 
> > > -Neil
> > > 
> > > > On Mar 27, 2015, at 11:11 AM, Chad notifications@github.com wrote:
> > > > 
> > > > Merge my branch with master
> > > > 
> > > > You can view, comment on, or merge this pull request online at:
> > > > 
> > > > https://github.com/faust06/Week4-Assn4/pull/6 <
> > > > https://github.com/faust06/Week4-Assn4/pull/6>
> > > > Commit Summary
> > > > 
> > > > Update common.h
> > > > Update HashADT.cpp
> > > > Update HashADT.h
> > > > Update OtherFunctions.cpp
> > > > Update OtherFunctions.h
> > > > derp add
> > > > Update main.cpp
> > > > Update Common.h
> > > > Update HashADT.cpp
> > > > Update HashADT.h
> > > > Update OtherFunctions.cpp
> > > > Update OtherFunctions.h
> > > > Update main.cpp
> > > > Included your functions and QuadraticProbe / DoubleHashValue
> > > > Added your functions abd QuadraticProbe / DoubleHashValue
> > > > 3 new consts added
> > > > File Changes
> > > > 
> > > > M townsendpeters-prog/Common.h <
> > > > https://github.com/faust06/Week4-Assn4/pull/6/files#diff-0> (24)
> > > > M townsendpeters-prog/HashADT.cpp <
> > > > https://github.com/faust06/Week4-Assn4/pull/6/files#diff-1> (278)
> > > > M townsendpeters-prog/HashADT.h <
> > > > https://github.com/faust06/Week4-Assn4/pull/6/files#diff-2> (8)
> > > > M townsendpeters-prog/OtherFunctions.cpp <
> > > > https://github.com/faust06/Week4-Assn4/pull/6/files#diff-3> (286)
> > > > M townsendpeters-prog/OtherFunctions.h <
> > > > https://github.com/faust06/Week4-Assn4/pull/6/files#diff-4> (7)
> > > > M townsendpeters-prog/main.cpp <
> > > > https://github.com/faust06/Week4-Assn4/pull/6/files#diff-5> (79)
> > > > Patch Links:
> > > > 
> > > > https://github.com/faust06/Week4-Assn4/pull/6.patch <
> > > > https://github.com/faust06/Week4-Assn4/pull/6.patch>
> > > > https://github.com/faust06/Week4-Assn4/pull/6.diff <
> > > > 
> > > > ## https://github.com/faust06/Week4-Assn4/pull/6.diff>
> > > > 
> > > > Reply to this email directly or view it on GitHub <
> > > > https://github.com/faust06/Week4-Assn4/pull/6>.
> > > 
> > > ## 
> > > 
> > > Reply to this email directly or view it on GitHub
> > > https://github.com/faust06/Week4-Assn4/pull/6#issuecomment-87037585.
> > > 
> > > ## 
> > > 
> > > Reply to this email directly or view it on GitHub.
> 
> ## 
> 
> Reply to this email directly or view it on GitHub
> https://github.com/faust06/Week4-Assn4/pull/6#issuecomment-87059686.

cpeters008 committed:
Added debug statements - testing Quadratic / Double Hash

-- Program now creates file with entire hash table so we can see the entire  contents for debugging / testing.
-- The code in the switch statements most likely won't stay and will be later transferred into a function to clean up main

cpeters008 committed:
Added another int to QuadraticProbe parameters (4 total param now)

Needed initialValue for comparison in function to prevent openHashTable array index #s being the same as the randomly generated number from the array.

cpeters008 committed:
Updated Quadratic Function

--array indexes no longer can be identical with the value
--still having some cluster issues, but its not as bad
--random numbers are random, but sometimes appear in numerical order ... I think this is because of the clock???
